ingenialink/servo.py

```diff
@@ -416,9 +415,12 @@ def save_configuration(self, config_file: str, subnode: Optional[int] = None) ->
         device.set("Interface", interface)
         if self.dictionary.part_number is not None:
             device.set("PartNumber", self.dictionary.part_number)
-        device.set("ProductCode", str(prod_code))
-        device.set("RevisionNumber", str(rev_number))
-        device.set("firmwareVersion", str(self.dictionary.firmware_version))
+        if self.dictionary.product_code is not None:
+            device.set("ProductCode", str(self.dictionary.product_code))
+        if self.dictionary.revision_number is not None:
+            device.set("RevisionNumber", str(self.dictionary.revision_number))
+        if self.dictionary.firmware_version is not None:
+            device.set("firmwareVersion", self.dictionary.firmware_version)
```

Comment:
I think these values should be read from the drive instead of set it as None

Reply:
Done